### DISCUSSION TITLE
How to load external msg.properties for MessageBundle?

### DISCUSSION THREAD

**Discussion by kinkkong** [quarkusio/quarkus#42123] [Q&A]
I'm using [Qute with Quarkus](https://quarkus.io/guides/qute-reference) and defined my `messages/msg.properties` file. It works as expected. What I'm trying to achive is to load `msg.properties` file from external configuration. 

Based on this article [Config Sources](https://quarkus.io/guides/config-reference#configuration-sources) when running native app I've added my properties file to `$PWD/config/` folder - so it is under `$PWD/config/messages/msg.properties`. Unfortunately this is still not loaded.

How could I achive to load properties from external source?

**Comment by michelkaeser** [comment]
Is there a solution to this?

**Comment by mkouba** [comment]
> How could I achive to load properties from external source?

It's not possible. Localized files are only loaded from application archives.

**Comment by michelkaeser** [comment]
As a workaround, it can in many situations be solved via Maven plugin:

```lang=xml
            <plugin>
                <groupId>org.apache.maven.plugins</groupId>
                <artifactId>maven-dependency-plugin</artifactId>
                <executions>
                    <execution>
                        <id>unpack-i18n-bundle</id> <!-- Quarkus' @MessageBundle -->
                        <phase>process-resources</phase>
                        <goals>
                            <goal>unpack</goal>
                        </goals>
                        <configuration>
                            <artifactItems>
                                <artifactItem>
                                    <groupId>...</groupId>
                                    <artifactId>i18n-bundle</artifactId>
                                    <overWrite>true</overWrite>
                                    <outputDirectory>${project.build.outputDirectory}</outputDirectory>
                                    <includes>messages/**</includes>
                                </artifactItem>
                            </artifactItems>
                        </configuration>
                    </execution>
                </executions>
            </plugin>
```

**Reply by mkouba** [reply]
But this assumes that localized files are stored in a dependency. That's not exactly an external source.
